bundles/org.openhab.binding.plugwiseha/src/main/java/org/openhab/binding/plugwiseha/internal/handler/PlugwiseHABaseHandler.java

```diff
@@ -0,0 +1,201 @@
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PlugwiseHABaseHandler} abstract class provides common methods and
+ * properties for the ThingHandlers of this binding. Extends @{link
+ * BaseThingHandler}
+ *
+ * @author Bas van Wetten - Initial contribution
+ * @author Leo Siepel - finish initial contribution
+ *
+ * @param <E> entity - the Plugwise Home Automation entity class used by this
+ *            thing handler
+ * @param <C> config - the Plugwise Home Automation config class used by this
+ *            thing handler
+ */
+
+@NonNullByDefault
+public abstract class PlugwiseHABaseHandler<E, C extends PlugwiseHAThingConfig> extends BaseThingHandler {
+
+    // Private Static error messages
+
+    protected static final String STATUS_DESCRIPTION_COMMUNICATION_ERROR = "Error communicating with the Plugwise Home Automation controller";
+
+    // private @Nullable C config;
+
+    protected final Logger logger = LoggerFactory.getLogger(PlugwiseHABaseHandler.class);
+
+    private Class<?> clazz;
+
+    // Constructor
+
+    public PlugwiseHABaseHandler(Thing thing) {
+        super(thing);
+        clazz = (Class<?>) (((ParameterizedType) getClass().getGenericSuperclass()).getActualTypeArguments()[1]);
+    }
+
+    // Abstract methods
+
+    protected abstract void initialize(C config);
+
+    protected abstract @Nullable E getEntity(PlugwiseHAController controller, Boolean forceRefresh)
+            throws PlugwiseHAException;
+
+    protected abstract void refreshChannel(E entity, ChannelUID channelUID);
+
+    protected abstract void handleCommand(E entity, ChannelUID channelUID, Command command) throws PlugwiseHAException;
+
+    // Overrides
+
+    @Override
+    public void initialize() {
+        C config = getPlugwiseThingConfig();
+
+        if (checkConfig(config)) {
+            // logger.debug("Initializing Plugwise Home Automation thing handler with config = {}", config);
+
+            Bridge bridge = getBridge();
+            if (bridge == null || bridge.getHandler() == null
+                    || !(bridge.getHandler() instanceof PlugwiseHABridgeHandler)) {
+                updateStatus(OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "You must choose a Plugwise Home Automation bridge for this thing.");
+                return;
+            }
+
+            if (bridge.getStatus() == OFFLINE) {
+                updateStatus(OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                        "The Plugwise Home Automation bridge is currently offline.");
+            }
+
+            initialize(config);
```

Comment:
If you pass the bridge to `initialize` and have a guaranteed non null object you don't need to check for null in the sub classes.

Reply:
That's better. I would also like to get `getPlugwiseHABridge()` to be non null, as that would take away many null checks in the subclasses. But because it uses `Bridge bridge = this.getBridge();` that is nullable, so what would the approach be to get this non null.